src/foam/u2/wizard/DaoWAO.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard',
+  name: 'DaoWAO',
+  implements: [
+    'foam.mlang.Expressions',
+    'foam.u2.wizard.WAO'
+  ],
+
+  requires: [
+    'foam.core.Slot',
+    'foam.mlang.expr.Dot',
+    'foam.core.Property',
+  ],
+  flags: ['web'],
+
+  properties: [
+    {
+      name: 'of',
+      class: 'Class'
+    },
+    {
+      name: 'daoKey',
+      class: 'String',
+      expression: function(of, path) {
+        if ( path ) {
+          const parts = path.split('.');
+          let of_ = of;
+          let obj = null
+          for ( let part of parts ) {
+            obj = of_.getAxiomByName(part);
+            of = obj.cls_;
+          }
+          return foam.String.daoize(obj.of.name)
+        }
+        return foam.String.daoize(of.name);
+      }
+    },
+    {
+      name: 'dao',
+      class: 'foam.dao.DAOProperty',
+      expression: function (daoKey) {
+        return this.__context__[daoKey];
+      }
+    },
+    {
+      class: 'String',
+      name: 'path',
+      documentation: `
+        Optional path used to specify the location of the desired wizardlet\'s data object that will
+        be updated.
+      `,
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'path_',
+      expression: function(path) {
+        if ( typeof path !== 'string' || path == '' ) return path;
+        const parts = path.split('.');
+        let expr = null;
+
+        for ( let part of parts ) {
+          const partialProperty = foam.core.Property.create({
+            name: part
+          });
+          expr = expr ? this.DOT(expr, partialProperty) : partialProperty;
+        }
+        return expr;
+      }
+    },
+    {
+      class: 'Boolean',
+      name: 'disableFind',
+      value: false
+    }
+  ],
+
+  methods: [
+    async function save(wizardlet) {
+      if ( wizardlet.loading ) return;
+      if ( ! wizardlet.isAvailable ) return;
+      wizardlet.loading = true;
+
+      let dataToPut = this.path_ ? this.path_.f(wizardlet.data) : wizardlet.data;
+
+      return await this.dao.put(dataToPut).then(savedData => {
+          this.setValue_(wizardlet, savedData);
+          return savedData;
+        })
+      .catch(e => {
+        console.error(e);
+        wizardlet.reportNetworkFailure(e);
+      })
+      .finally(() => {
+        wizardlet.loading = false;
+      });
+    },
+    async function load(wizardlet) {
+      if ( wizardlet.loading ) return;
+      if ( ! this.disableFind ) {
```

Comment:
Is the disableFind case for when we want to use the DAOWAO in a create case?

Reply:
yeah, when we want to restrict the DAOWAO for only the create case.